### PR TITLE
fix Photon OS mounting issue RAC-4417

### PR DESCRIPTION
**Pull request by xuea1**
fix Photon OS mounting issue RAC-4417

### CODE REVIEW

**Comment by iceiilin** [comment]
Is there necessary to remove the same check here https://github.com/xuea1/on-tools/blob/e50485aea6d188366711b788c95306fbb3ae5d98/scripts/setup_iso.py#L171 ?

File: scripts/setup_iso.py [hunk 1]
```diff
@@ -153,6 +153,9 @@ def determine_os_ver(srcdir, iso_info):
             print 'attempting LiveCD netboot'
             osname = 'LiveCD'
             osver = vid
+            for i in src_dir_list:
+                if 'PHOTON' in i:
+                    osname = 'PHOTON'
```
Comment (lanchongyizu) [inline scripts/setup_iso.py]: It's better TO
Change
`elif 'isolinux' in src_dir_list:`
to
`elif 'LiveOS' in src_dir_list and 'isolinux' in src_dir_list:`

And change
```
else:
    for i in src_dir_list:
        if "PHOTON" in i:
            osname = "PHOTON"
            if vid == '':
                assert ValueError, "The volume ID is not get correctly"
            osver = vid.lstrip('PHOTON_')
```
to
```
elif 'PHOTON_VMWARE_CD' in src_dir_list:
    osname = "PHOTON"
    if vid == '':
        assert ValueError, "The volume ID is not get correctly"
    osver = vid.lstrip('PHOTON_')
```

**Comment by xuea1** [comment]
@iceiilin my intension of those changes is meant to fix RAC-4417, which is about Photon iso image contain 'isolinux' after mount into system, and in the same time, it also contain ''PHOTON_VMWARE_CD' after mount into system. so if above 2 condition met, I deem it as Photon os, instead of 'LiveCD'.

Refer to code between Line #171 and #176, I think someone must have verified some Photon iso images, and meet this type of condition, so I just leave them as they were.

File: scripts/setup_iso.py (hunk 1 above)
Comment (xuea1) [inline scripts/setup_iso.py]: are you suggesting code as below?

[...]
      elif 'LiveCD' in src_dir_list and 'isolinux' in src_dir_list: 
            print 'attempting LiveCD netboot'
            osname = 'LiveCD'
            osver = vid

      elif 'suse' in src_dir_list:
            osname = 'SUSE'
            if vid == '':
                assert ValueError, "The volume ID is not get correctly"
            ver_pattern = '(.+)-DVD-(.+)'
            m = re.search(ver_pattern, vid)
            assert m.group(1), "The version name is not retrieved correctly"
            osver = m.group(1)
        elif 'setup.exe' in src_dir_list:
            osname, osver = get_setup_exe_name_version(srcdir)
        elif 'PHOTON_VMWARE_CD' in src_dir_list:
            osname = "PHOTON"
            if vid == '':
               assert ValueError, "The volume ID is not get correctly"
           osver = vid.lstrip('PHOTON_')
[...]

Comment (xuea1) [inline scripts/setup_iso.py]: are you suggesting code as below?

[...]

      elif 'LiveCD' in src_dir_list and 'isolinux' in src_dir_list: 
            print 'attempting LiveCD netboot'
            osname = 'LiveCD'
            osver = vid

      elif 'suse' in src_dir_list:
            osname = 'SUSE'
            if vid == '':
                assert ValueError, "The volume ID is not get correctly"
            ver_pattern = '(.+)-DVD-(.+)'
            m = re.search(ver_pattern, vid)
            assert m.group(1), "The version name is not retrieved correctly"
            osver = m.group(1)
        elif 'setup.exe' in src_dir_list:
            osname, osver = get_setup_exe_name_version(srcdir)
        elif 'PHOTON_VMWARE_CD' in src_dir_list:
            osname = "PHOTON"
            if vid == '':
               assert ValueError, "The volume ID is not get correctly"
           osver = vid.lstrip('PHOTON_')
[...]

Comment (xuea1) [inline scripts/setup_iso.py]: are you suggesting code as below?

[...]

      elif 'LiveCD' in src_dir_list and 'isolinux' in src_dir_list: 
            print 'attempting LiveCD netboot'
            osname = 'LiveCD'
            osver = vid

      elif 'suse' in src_dir_list:
            osname = 'SUSE'
            if vid == '':
                assert ValueError, "The volume ID is not get correctly"
            ver_pattern = '(.+)-DVD-(.+)'
            m = re.search(ver_pattern, vid)
            assert m.group(1), "The version name is not retrieved correctly"
            osver = m.group(1)
        elif 'setup.exe' in src_dir_list:
            osname, osver = get_setup_exe_name_version(srcdir)
        elif 'PHOTON_VMWARE_CD' in src_dir_list:
            osname = "PHOTON"
            if vid == '':
               assert ValueError, "The volume ID is not get correctly"
           osver = vid.lstrip('PHOTON_')
[...]

Comment (xuea1) [inline scripts/setup_iso.py]: are you suggesting code as below?

[...]

      elif 'LiveCD' in src_dir_list and 'isolinux' in src_dir_list: 
            print 'attempting LiveCD netboot'
            osname = 'LiveCD'
            osver = vid

      elif 'suse' in src_dir_list:
            osname = 'SUSE'
            if vid == '':
                assert ValueError, "The volume ID is not get correctly"
            ver_pattern = '(.+)-DVD-(.+)'
            m = re.search(ver_pattern, vid)
            assert m.group(1), "The version name is not retrieved correctly"
            osver = m.group(1)
        elif 'setup.exe' in src_dir_list:
            osname, osver = get_setup_exe_name_version(srcdir)
        elif 'PHOTON_VMWARE_CD' in src_dir_list:
            osname = "PHOTON"
            if vid == '':
               assert ValueError, "The volume ID is not get correctly"
           osver = vid.lstrip('PHOTON_')
[...]

**Comment by xuea1** [comment]
@lanchongyizu 
are you suggesting code as below?

[...]

      elif 'LiveCD' in src_dir_list and 'isolinux' in src_dir_list: 
            print 'attempting LiveCD netboot'
            osname = 'LiveCD'
            osver = vid

      elif 'suse' in src_dir_list:
            osname = 'SUSE'
            if vid == '':
                assert ValueError, "The volume ID is not get correctly"
            ver_pattern = '(.+)-DVD-(.+)'
            m = re.search(ver_pattern, vid)
            assert m.group(1), "The version name is not retrieved correctly"
            osver = m.group(1)
        elif 'setup.exe' in src_dir_list:
            osname, osver = get_setup_exe_name_version(srcdir)
        elif 'PHOTON_VMWARE_CD' in src_dir_list:
            osname = "PHOTON"
            if vid == '':
               assert ValueError, "The volume ID is not get correctly"
           osver = vid.lstrip('PHOTON_')
[...]

File: scripts/setup_iso.py (hunk 1 above)
Comment (yyscamper) [inline scripts/setup_iso.py]: need break if found "PHOTON"

**Comment by xuea1** [comment]
revert LiveCD OS name related code according to history which can fix Photon OS mounting issue RAC-4417

File: scripts/setup_iso.py (hunk 1 above)
Comment (iceiilin) [inline scripts/setup_iso.py]: what about
```
if "PHOTON" in src_dir_list:
    osname = 'PHOTON'
```

?

**Pull request merged**